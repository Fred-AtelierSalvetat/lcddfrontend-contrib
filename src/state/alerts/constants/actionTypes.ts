export const NEW_ALERT = 'NEW_ALERT' as const;
export const DISMISS_ALERT = 'DISMISS_ALERT' as const;
