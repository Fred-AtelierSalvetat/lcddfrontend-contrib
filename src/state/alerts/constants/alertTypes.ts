export const SUCCESS = 'SUCCESS' as const;
export const FAILURE = 'FAILURE' as const;

export type AlertsType = typeof SUCCESS | typeof FAILURE;
