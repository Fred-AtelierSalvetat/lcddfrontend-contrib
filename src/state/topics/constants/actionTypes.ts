export const FETCH_TOPICS_REQUEST = 'FETCH_TOPICS_REQUEST' as const;
export const FETCH_TOPICS_SUCCESS = 'FETCH_TOPICS_SUCCESS' as const;
export const FETCH_TOPICS_FAILURE = 'FETCH_TOPICS_FAILURE' as const;

export type TopicActionType = typeof FETCH_TOPICS_REQUEST | typeof FETCH_TOPICS_SUCCESS | typeof FETCH_TOPICS_FAILURE;
