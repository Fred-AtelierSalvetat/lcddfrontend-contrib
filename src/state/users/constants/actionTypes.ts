export const FETCH_USERS_REQUEST = 'FETCH_USERS_REQUEST' as const;
export const FETCH_USERS_SUCCESS = 'FETCH_USERS_SUCCESS' as const;
export const FETCH_USERS_FAILURE = 'FETCH_USERS_FAILURE' as const;
export const ACTIVATE_USER_REQUEST = 'ACTIVATE_USER_REQUEST' as const;
export const ACTIVATE_USER_SUCCESS = 'ACTIVATE_USER_SUCCESS' as const;
export const ACTIVATE_USER_FAILURE = 'ACTIVATE_USER_FAILURE' as const;
export const DEACTIVATE_USER_REQUEST = 'DEACTIVATE_USER_REQUEST' as const;
export const DEACTIVATE_USER_SUCCESS = 'DEACTIVATE_USER_SUCCESS' as const;
export const DEACTIVATE_USER_FAILURE = 'DEACTIVATE_USER_FAILURE' as const;
export const DELETE_USER_REQUEST = 'DELETE_USER_REQUEST' as const;
export const DELETE_USER_SUCCESS = 'DELETE_USER_SUCCESS' as const;
export const DELETE_USER_FAILURE = 'DELETE_USER_FAILURE' as const;
export const PROMOTE_USER_TO_ADMIN_REQUEST = 'PROMOTE_USER_TO_ADMIN_REQUEST' as const;
export const PROMOTE_USER_TO_ADMIN_SUCCESS = 'PROMOTE_USER_TO_ADMIN_SUCCESS' as const;
export const PROMOTE_USER_TO_ADMIN_FAILURE = 'PROMOTE_USER_TO_ADMIN_FAILURE' as const;
export const REVOKE_USER_ADMIN_RIGHT_REQUEST = 'REVOKE_USER_ADMIN_RIGHT_REQUEST' as const;
export const REVOKE_USER_ADMIN_RIGHT_SUCCESS = 'REVOKE_USER_ADMIN_RIGHT_SUCCESS' as const;
export const REVOKE_USER_ADMIN_RIGHT_FAILURE = 'REVOKE_USER_ADMIN_RIGHT_FAILURE' as const;
export const PROMOTE_USER_TO_SPEAKER_REQUEST = 'PROMOTE_USER_TO_SPEAKER_REQUEST' as const;
export const PROMOTE_USER_TO_SPEAKER_SUCCESS = 'PROMOTE_USER_TO_SPEAKER_SUCCESS' as const;
export const PROMOTE_USER_TO_SPEAKER_FAILURE = 'PROMOTE_USER_TO_SPEAKER_FAILURE' as const;
export const VALIDATE_SPEAKER_REQUEST = 'VALIDATE_SPEAKER_REQUEST' as const;
export const VALIDATE_SPEAKER_SUCCESS = 'VALIDATE_SPEAKER_SUCCESS' as const;
export const VALIDATE_SPEAKER_FAILURE = 'VALIDATE_SPEAKER_FAILURE' as const;

export const SET_USER_SEARCH_FILTER = 'SET_USER_SEARCH_FILTER' as const;
export const SET_USER_ROLE_FILTER = 'SET_USER_ROLE_FILTER' as const;

export type UsersActionType =
    | typeof FETCH_USERS_REQUEST
    | typeof FETCH_USERS_SUCCESS
    | typeof FETCH_USERS_FAILURE
    | typeof ACTIVATE_USER_REQUEST
    | typeof ACTIVATE_USER_SUCCESS
    | typeof ACTIVATE_USER_FAILURE
    | typeof DEACTIVATE_USER_REQUEST
    | typeof DEACTIVATE_USER_SUCCESS
    | typeof DEACTIVATE_USER_FAILURE
    | typeof DELETE_USER_REQUEST
    | typeof DELETE_USER_SUCCESS
    | typeof DELETE_USER_FAILURE
    | typeof PROMOTE_USER_TO_ADMIN_REQUEST
    | typeof PROMOTE_USER_TO_ADMIN_SUCCESS
    | typeof PROMOTE_USER_TO_ADMIN_FAILURE
    | typeof REVOKE_USER_ADMIN_RIGHT_REQUEST
    | typeof REVOKE_USER_ADMIN_RIGHT_SUCCESS
    | typeof REVOKE_USER_ADMIN_RIGHT_FAILURE
    | typeof PROMOTE_USER_TO_SPEAKER_REQUEST
    | typeof PROMOTE_USER_TO_SPEAKER_SUCCESS
    | typeof PROMOTE_USER_TO_SPEAKER_FAILURE
    | typeof VALIDATE_SPEAKER_REQUEST
    | typeof VALIDATE_SPEAKER_SUCCESS
    | typeof VALIDATE_SPEAKER_FAILURE
    | typeof SET_USER_SEARCH_FILTER
    | typeof SET_USER_ROLE_FILTER;
