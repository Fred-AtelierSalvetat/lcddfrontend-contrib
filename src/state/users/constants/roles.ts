export const ROLE_ADMIN = 'Admin';
export const ROLE_SPEAKER_AWAITING_ANSWER = 'Intervenant, en attente';
export const ROLE_SPEAKER_AWAITING_VALIDATION = 'Intervenant, Ã  valider';
export const ROLE_SPEAKER = 'Intervenant';
export const ROLE_PRO_USER = 'Pro du droit';
export const ROLE_STUDENT = 'Etudiant';
export const ROLE_CITIZEN = 'Citoyen';

export const ADMIN_ROLE_KEY = 'admin';
export const SPEAKER_ROLE_KEY = 'speaker';
export const USER_ROLE_KEY = 'user';

const ADMIN_TAB_ROLE_FILTER = [ROLE_ADMIN];
const SPEAKER_TAB_ROLE_FILTER = [ROLE_SPEAKER_AWAITING_ANSWER, ROLE_SPEAKER_AWAITING_VALIDATION, ROLE_SPEAKER];
const USERS_TAB_ROLE_FILTER = [
    ROLE_PRO_USER,
    ROLE_STUDENT,
    ROLE_CITIZEN,
    ROLE_SPEAKER_AWAITING_ANSWER,
    ROLE_SPEAKER_AWAITING_VALIDATION,
    ROLE_SPEAKER,
];

export const roleFilterMap = {
    [ADMIN_ROLE_KEY]: ADMIN_TAB_ROLE_FILTER,
    [SPEAKER_ROLE_KEY]: SPEAKER_TAB_ROLE_FILTER,
    [USER_ROLE_KEY]: USERS_TAB_ROLE_FILTER,
};
