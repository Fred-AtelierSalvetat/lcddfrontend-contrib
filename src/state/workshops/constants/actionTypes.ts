export const FETCH_WORKSHOPS = 'FETCH_WORKSHOPS' as const;
export const CREATE_WORKSHOP = 'CREATE_WORKSHOP' as const;
export const UPDATE_WORKSHOP = 'UPDATE_WORKSHOP' as const;
export const CANCEL_WORKSHOP = 'CANCEL_WORKSHOP' as const;
export const DELETE_WORKSHOP = 'DELETE_WORKSHOP' as const;
export const ARCHIVE_WORKSHOP = 'ARCHIVE_WORKSHOP' as const;
export const RESTORE_WORKSHOP = 'RESTORE_WORKSHOP' as const;
export const GO_LIVE_WORKSHOP = 'GO_LIVE_WORKSHOP' as const;
export const END_LIVE_WORKSHOP = 'END_LIVE_WORKSHOP' as const;
export const SET_WORKSHOP_SEARCH_FILTER = 'SET_WORKSHOP_SEARCH_FILTER' as const;
export const SET_ORDER_BY = 'SET_ORDER_BY' as const;

export type WorkshopsActionType =
    | typeof FETCH_WORKSHOP
    | typeof CREATE_WORKSHOP
    | typeof UPDATE_WORKSHOP
    | typeof CANCEL_WORKSHOP
    | typeof DELETE_WORKSHOP
    | typeof ARCHIVE_WORKSHOP
    | typeof RESTORE_WORKSHOP
    | typeof GO_LIVE_WORKSHOP
    | typeof END_LIVE_WORKSHOP
    | typeof SET_WORKSHOP_SEARCH_FILTER
    | typeof SET_ORDER_BY;
