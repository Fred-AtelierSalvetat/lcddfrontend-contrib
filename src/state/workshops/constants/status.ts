export const INCOMING = 'A venir' as const;
export const LIVE = 'EN DIRECT' as const;
export const UNPUBLISHED = 'Inédit' as const;
export const PUBLISHED = 'Publié' as const;
export const ARCHIVED = 'Archivé' as const;

export const statusOrder = [INCOMING, LIVE, UNPUBLISHED, PUBLISHED, ARCHIVED];
export type WorkshopStatusType = typeof INCOMING | typeof LIVE | typeof PUBLISHED | typeof ARCHIVED;
