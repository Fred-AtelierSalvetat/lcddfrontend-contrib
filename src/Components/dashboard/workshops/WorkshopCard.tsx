import React, { FC } from 'react';
import { useDispatch } from 'react-redux';
import { useHistory } from 'react-router-dom';
import Card from 'react-bootstrap/Card';
import className from 'classnames';
import PropTypes from 'prop-types';
import format from 'date-fns/format';
import { fr } from 'date-fns/locale';

import { ReactComponent as EditIcon } from '~/assets/icons/edit_24px.svg';
import { ReactComponent as MoreVertIcon } from '~/assets/icons/more_vert_24px.svg';
import { ReactComponent as DeleteForeverIcon } from '~/assets/icons/delete_forever_16px.svg';
import { ReactComponent as CancelIcon } from '~/assets/icons/cancel_24px.svg';
import { ReactComponent as BackupRestoreIcon } from '~/assets/icons/settings_backup_restore_24px.svg';
import { ReactComponent as VideoCamIcon } from '~/assets/icons/videocam_24px.svg';

import ActionMenuPopover from '~/Components/shared/actionMenuPopover/ActionMenuPopover';
import Action from '~/Components/shared/actionMenuPopover/Action';
import ConfirmDialog from '~/Components/shared/modals/ConfirmDialog';
import * as statusConst from '~/state/workshops/constants/status';
import * as action from '~/state/workshops/actions';

import './WorkshopCard.scss';

const propTypes = {
    workshop: PropTypes.shape({
        id: PropTypes.string.isRequired,
        status: PropTypes.string.isRequired,
        thumbnail: PropTypes.string,
        title: PropTypes.string.isRequired,
        startingdate: PropTypes.object.isRequired,
        // duration: PropTypes.object,
        topics: PropTypes.arrayOf(PropTypes.string).isRequired,
        description: PropTypes.string.isRequired,
    }),
};

const WorkshopCard: FC<PropTypes.InferProps<typeof propTypes>> = ({
    workshop: { id, status, thumbnail, title, startingdate, topics, description },
}) => {
    const dispatch = useDispatch();
    const history = useHistory();

    return (
        <Card className="card">
            <div className="image-container">
                {!!thumbnail && (
                    // && <Card.Img variant="top" alt="Workshop's thumbnail" src={thumbnail} />}
                    <Card.Img variant="top" alt="Workshop's thumbnail" data-src={thumbnail} className="lazyload" />
                )}
                <Card.ImgOverlay>
                    <div
                        className={className('status', {
                            incoming: status === statusConst.INCOMING,
                            live: status === statusConst.LIVE,
                            published: status === statusConst.PUBLISHED,
                            unpublished: status === statusConst.UNPUBLISHED,
                            archived: status === statusConst.ARCHIVED,
                        })}
                    >
                        {status}
                    </div>
                    {status !== statusConst.LIVE && (
                        <div className="actionMenuContainer">
                            <ActionMenuPopover icon={<MoreVertIcon title="Actions" />} placement="bottom-end">
                                {status === statusConst.INCOMING && (
                                    <>
                                        <Action
                                            icon={<CancelIcon />}
                                            label="Annuler"
                                            modalConfirmation={
                                                <ConfirmDialog
                                                    show
                                                    title="Annuler cet atelier"
                                                    body="Cette action n’est pas réversible."
                                                    cancelButton="Annuler"
                                                    okButton="Annuler l'atelier"
                                                    handleClose={() => {}}
                                                    handleConfirm={() => dispatch(action.cancelWorkshop(id))}
                                                />
                                            }
                                        />
                                        <Action
                                            icon={<VideoCamIcon />}
                                            label="GO LIVE"
                                            action={() => history.push(`/dashboard/goLive/${id}`)}
                                        />
                                    </>
                                )}
                                {status === statusConst.PUBLISHED && (
                                    <Action
                                        icon={<BackupRestoreIcon />}
                                        label="Archiver"
                                        action={() => dispatch(action.archiveWorkshop(id))}
                                    />
                                )}
                                {status === statusConst.ARCHIVED && (
                                    <Action
                                        icon={<BackupRestoreIcon />}
                                        label="Restaurer"
                                        action={() => dispatch(action.restoreWorkshop(id))}
                                    />
                                )}
                                {[statusConst.PUBLISHED, statusConst.UNPUBLISHED, statusConst.ARCHIVED].includes(
                                    status,
                                ) && (
                                    <Action
                                        icon={<DeleteForeverIcon />}
                                        label="Supprimer"
                                        modalConfirmation={
                                            <ConfirmDialog
                                                show
                                                title="Supprimer cet atelier et toutes les videos associées"
                                                body="Cette action n’est pas réversible."
                                                cancelButton="Annuler"
                                                okButton="Supprimer"
                                                handleClose={() => {}}
                                                handleConfirm={() => dispatch(action.deleteWorkshop(id))}
                                            />
                                        }
                                    />
                                )}
                            </ActionMenuPopover>
                        </div>
                    )}
                    <EditIcon
                        title="Modifier"
                        className="editIcon"
                        onClick={() => {
                            history.push(`/dashboard/editWorkshop/${id}`);
                        }}
                    />
                    {status === statusConst.INCOMING ? (
                        <div className="startingdate">{format(startingdate, 'd MMMM yyyy HH:mm', { locale: fr })}</div>
                    ) : (
                        <div className="duration">14:00:45</div> // TODO duration
                    )}
                </Card.ImgOverlay>
            </div>

            <Card.Body>
                <Card.Title>{title}</Card.Title>
                <Card.Text className="topics">{topics.join(', ')}</Card.Text>
                <Card.Text className="description">{description}</Card.Text>
            </Card.Body>
        </Card>
    );
};

WorkshopCard.propTypes = propTypes;

export default WorkshopCard;
