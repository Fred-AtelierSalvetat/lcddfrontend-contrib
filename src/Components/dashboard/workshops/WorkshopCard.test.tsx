import React from 'react';
import { mount } from 'enzyme';
import { screen, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { MemoryRouter, Route } from 'react-router-dom';
import format from 'date-fns/format';
import { fr } from 'date-fns/locale';

import WorkshopCard from './WorkshopCard';

import * as status from '~/state/workshops/constants/status';

describe('<WorkshopChard /> ', () => {
  const workshop = {
    id: '0',
    status: status.INCOMING,
    thumbnail: '',
    video: '',
    title: 'Le droit et le dev',
    startingdate: new Date('2021-07-20 12:34:56'),
    speakers: '',
    topics: ['TopicA', 'TopicB'],
    refsLegifrance: [],
    description: 'WorkshopDescriptionString',
    keywords: [],
    files: [],
    links: [],
  };

  const store = configureMockStore()({});

  it('renders without crashing', () => {
    render(
            <Provider store={store}>
                <WorkshopCard workshop={workshop} />
            </Provider>,
    );
  });

  it('shall render workshop card', () => {
    const { getByText } = render(
            <Provider store={store}>
                <WorkshopCard workshop={workshop} />
            </Provider>,
    );
    expect(getByText(workshop.title)).toBeInTheDocument();
    expect(getByText(format(workshop.startingdate, 'd MMMM yyyy HH:mm', { locale: fr }))).toBeInTheDocument();
    expect(getByText(workshop.description)).toBeInTheDocument();
    workshop.topics.forEach((topic) => expect(getByText(new RegExp(topic))).toBeInTheDocument());
  });

  it('shall allow CANCEL_WORKSHOP and GO_LIVE_WORKSHOP actions on INCOMING workshops', async () => {
    const { getByTitle } = render(
            <Provider store={store}>
                <WorkshopCard workshop={{ ...workshop, status: status.INCOMING }} />
            </Provider>,
    );
    userEvent.click(getByTitle('Actions'));
    await screen.findByText('Annuler');
    await screen.findByText('GO LIVE');
  });
  it('shall not allow any action on LIVE workshops', () => {
    const { queryByTitle } = render(
            <Provider store={store}>
                <WorkshopCard workshop={{ ...workshop, status: status.LIVE }} />
            </Provider>,
    );
    expect(queryByTitle('Actions')).not.toBeInTheDocument();
  });
  it('shall allow DELETE_WORKSHOP action on UNPUBLISHED workshops', async () => {
    const { getByTitle } = render(
            <Provider store={store}>
                <WorkshopCard workshop={{ ...workshop, status: status.UNPUBLISHED }} />
            </Provider>,
    );
    userEvent.click(getByTitle('Actions'));
    await screen.findByText('Supprimer');
  });

  it('shall allow ARCHIVE_WORKSHOP and DELETE_WORKSHOP actions on PUBLISHED workshops', async () => {
    const { getByTitle } = render(
            <Provider store={store}>
                <WorkshopCard workshop={{ ...workshop, status: status.PUBLISHED }} />
            </Provider>,
    );
    userEvent.click(getByTitle('Actions'));
    await screen.findByText('Archiver');
    await screen.findByText('Supprimer');
  });

  it('shall allow RESTORE_WORKSHOP and DELETE_WORKSHOP actions on ARCHIVED workshops', async () => {
    const { getByTitle } = render(
            <Provider store={store}>
                <WorkshopCard workshop={{ ...workshop, status: status.ARCHIVED }} />
            </Provider>,
    );
    userEvent.click(getByTitle('Actions'));
    await screen.findByText('Restaurer');
    await screen.findByText('Supprimer');
  });
  it('shall allow workshop edition', async () => {
    const { getByTitle } = render(
            <Provider store={store}>
                <MemoryRouter initialEntries={['/workshop']}>
                    <Route path="/workshop">
                        <WorkshopCard workshop={workshop} />
                    </Route>
                    <Route path={`/dashboard/editWorkshop/${workshop.id}`}>
                        <div>EditionPage</div>
                    </Route>
                </MemoryRouter>
            </Provider>,
    );
    userEvent.click(getByTitle('Modifier'));
    await screen.findByText('EditionPage');
  });
  it('should match its reference snapshot', () => {
    const wrapper = mount(
            <Provider store={store}>
                <WorkshopCard workshop={workshop} />
            </Provider>,
    );
    expect(wrapper).to.matchSnapshot();
  });
});
