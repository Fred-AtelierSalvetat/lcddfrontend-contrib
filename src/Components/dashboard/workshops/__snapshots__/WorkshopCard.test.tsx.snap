// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WorkshopChard />  should match its reference snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <WorkshopCard
    workshop={
      Object {
        "description": "WorkshopDescriptionString",
        "files": Array [],
        "id": "0",
        "keywords": Array [],
        "links": Array [],
        "refsLegifrance": Array [],
        "speakers": "",
        "startingdate": 2021-07-20T10:34:56.000Z,
        "status": "A venir",
        "thumbnail": "",
        "title": "Le droit et le dev",
        "topics": Array [
          "TopicA",
          "TopicB",
        ],
        "video": "",
      }
    }
  >
    <Card
      body={false}
      className="card"
    >
      <div
        className="card card"
      >
        <div
          className="image-container"
        >
          <CardImgOverlay>
            <div
              className="card-img-overlay"
            >
              <div
                className="status incoming"
              >
                A venir
              </div>
              <div
                className="actionMenuContainer"
              >
                <ActionMenuPopover
                  icon={
                    <ForwardRef(SvgMoreVert24Px)
                      title="Actions"
                    />
                  }
                  placement="bottom-end"
                >
                  <OverlayTrigger
                    defaultShow={false}
                    overlay={
                      <ForwardRef
                        id="ActionMenuPopover"
                        placement="right"
                      >
                        <ForwardRef>
                          <React.Fragment>
                            <Action
                              icon={<ForwardRef(SvgCancel24Px) />}
                              label="Annuler"
                              modalConfirmation={
                                <ConfirmDialog
                                  body="Cette action n’est pas réversible."
                                  cancelButton="Annuler"
                                  handleClose={[Function]}
                                  handleConfirm={[Function]}
                                  okButton="Annuler l'atelier"
                                  show={true}
                                  title="Annuler cet atelier"
                                />
                              }
                            />
                            <Action
                              action={[Function]}
                              icon={<ForwardRef(SvgVideocam24Px) />}
                              label="GO LIVE"
                            />
                          </React.Fragment>
                        </ForwardRef>
                      </ForwardRef>
                    }
                    placement="bottom-end"
                    rootClose={true}
                    trigger="click"
                  >
                    <RefHolder>
                      <div
                        className="ActionMenuIcon"
                        onClick={[Function]}
                      >
                        <ForwardRef(SvgMoreVert24Px)
                          title="Actions"
                        >
                          <svg
                            title="Actions"
                          >
                            more_vert_24px.svg
                          </svg>
                        </ForwardRef(SvgMoreVert24Px)>
                      </div>
                    </RefHolder>
                    <Overlay
                      flip={false}
                      onHide={[Function]}
                      placement="bottom-end"
                      popperConfig={Object {}}
                      rootClose={true}
                      show={false}
                      target={[Function]}
                      transition={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "defaultProps": Object {
                            "appear": false,
                            "in": false,
                            "mountOnEnter": false,
                            "timeout": 300,
                            "unmountOnExit": false,
                          },
                          "render": [Function],
                        }
                      }
                    >
                      <Overlay
                        flip={false}
                        onHide={[Function]}
                        placement="bottom-end"
                        popperConfig={
                          Object {
                            "modifiers": Array [
                              Object {
                                "name": "offset",
                                "options": Object {
                                  "offset": [Function],
                                },
                              },
                              Object {
                                "effect": [Function],
                                "enabled": true,
                                "name": "popoverArrowMargins",
                                "phase": "main",
                                "requiresIfExists": Array [
                                  "arrow",
                                ],
                              },
                            ],
                          }
                        }
                        rootClose={true}
                        show={false}
                        target={[Function]}
                        transition={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "defaultProps": Object {
                              "appear": false,
                              "in": false,
                              "mountOnEnter": false,
                              "timeout": 300,
                              "unmountOnExit": false,
                            },
                            "render": [Function],
                          }
                        }
                      />
                    </Overlay>
                  </OverlayTrigger>
                </ActionMenuPopover>
              </div>
              <ForwardRef(SvgEdit24Px)
                className="editIcon"
                onClick={[Function]}
                title="Modifier"
              >
                <svg
                  className="editIcon"
                  onClick={[Function]}
                  title="Modifier"
                >
                  edit_24px.svg
                </svg>
              </ForwardRef(SvgEdit24Px)>
              <div
                className="startingdate"
              >
                20 juillet 2021 12:34
              </div>
            </div>
          </CardImgOverlay>
        </div>
        <CardBody>
          <div
            className="card-body"
          >
            <CardTitle>
              <ForwardRef
                className="card-title"
              >
                <div
                  className="card-title h5"
                >
                  Le droit et le dev
                </div>
              </ForwardRef>
            </CardTitle>
            <CardText
              className="topics"
            >
              <p
                className="topics card-text"
              >
                TopicA, TopicB
              </p>
            </CardText>
            <CardText
              className="description"
            >
              <p
                className="description card-text"
              >
                WorkshopDescriptionString
              </p>
            </CardText>
          </div>
        </CardBody>
      </div>
    </Card>
  </WorkshopCard>
</Provider>
`;
